src/main/java/com/github/cameltooling/lsp/internal/parser/ParserFileHelperFactory.java

```diff
@@ -119,16 +111,75 @@ private boolean isCamelKYamlDSL(TextDocumentItem textDocumentItem, String uri) {
 	}
 
 	private boolean isYamlFileWithCamelKModelineLike(TextDocumentItem textDocumentItem, String uri) {
-		return uri.endsWith(".yaml") && textDocumentItem.getText().startsWith(CamelKModelineParser.MODELINE_LIKE_CAMEL_K_YAML);
+		return uri.endsWith(YAML_FILENAME_SUFFIX) && textDocumentItem.getText().startsWith(CamelKModelineParser.MODELINE_LIKE_CAMEL_K_YAML);
 	}
 
 	protected boolean isYamlFileWithCamelKShebang(TextDocumentItem textDocumentItem, String uri) {
-		return uri.endsWith(".yaml") && textDocumentItem.getText().startsWith(SHEBANG_CAMEL_K);
+		return uri.endsWith(YAML_FILENAME_SUFFIX) && textDocumentItem.getText().startsWith(SHEBANG_CAMEL_K);
 	}
 
 	private boolean isCamelJavaDSL(TextDocumentItem textDocumentItem, String uri) {
 		//improve this method to provide better heuristic to detect if it is a Camel file or not
-		return uri.endsWith(".java") && textDocumentItem.getText().contains("camel");
+		return uri.endsWith(JAVA_FILENAME_SUFFIX) && textDocumentItem.getText().contains("camel");
 	}
 
+	private boolean isCamelXmlDSL(String uri) {
+		return uri.endsWith(XML_FILENAME_SUFFIX);
+	}
```

Comment:
it is no more checking for `xmlParser.getCorrespondingCamelNodeForCompletion(textDocumentItem, line) != null`